### ISSUE TITLE
Сustom authorization

### ISSUE THREAD

**Issue by wwwhttpru** [apstanisic/directus-dart#21]
My projects use custom authorization, this library does not allow me to add my own authorization.
Can we think of something?
I also want to organize this library!

**Comment by wwwhttpru** [comment]
Pardon my english hereafter :)
I see that you are checking user authorization, depending on the presence of a token.
https://github.com/apstanisic/directus-dart/blob/62e151a387dbb596dfc91fb5a8968db180fe1312/lib/src/modules/auth/auth_handler.dart#L80-L93

And after the action "logout" you do not clear the local storage
https://github.com/apstanisic/directus-dart/blob/62e151a387dbb596dfc91fb5a8968db180fe1312/lib/src/modules/auth/auth_handler.dart#L126-L140

I have not tested on a real project, but I can assume that after exiting the application and restarting, the user will be logged in. Correct me if I'm wrong!

Thank you for the attention!

**Comment by apstanisic** [comment]
You are right, it's not removing data from `shared_preferences`. It's a bug. Would you like to create a PR?

You can access Dio instance with `sdk.client`, and that allows you to add custom headers and interceptors, and send requests.

**Comment by wwwhttpru** [comment]
I have ideas for implementing authorization following the example of Firebase. May I suggest my own version for consideration?

**Comment by apstanisic** [comment]
This library is only providing methods for Directus API endpoints, there are no other features that could be implemented. What would you do different? 

There is unstable `onChange` setter that can be used similarly as Firebase.
```dart
sdk.auth.onChange = (String actionType, AuthResponse? data) {
  if (actionType === 'logout') {
    print('Logout');
  }
}
```

**Comment by wwwhttpru** [comment]
Here is an example of my idea
The user is logged in when user != null
```dart
_subscription = widget.sdk.auth.user.listen((user) {
   setState(() {
       user = user;
    });
});
```

**Comment by wwwhttpru** [comment]
I opened the PR, I will be glad if you look #22

**Comment by apstanisic** [comment]
I'll check it out soon

**Comment by cordschneider** [comment]
Any progress on reviewing PR #22?

**Comment by apstanisic** [comment]
Closed with #25
